client/vite.config.js

```diff
@@ -0,0 +1,15 @@
+import { fileURLToPath, URL } from "node:url";
+
+import { defineConfig } from "vite";
+import vue from "@vitejs/plugin-vue";
+import vueJsx from "@vitejs/plugin-vue-jsx";
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  plugins: [vue(), vueJsx()],
+  resolve: {
+    alias: {
+      "@": fileURLToPath(new URL("./src", import.meta.url)),
```

Comment:
I can only try 😁 There are no good resources that fully explain this so I broke it down into smaller pieces and connected everything. I'll explain things from inside out.

So, `import.meta.url` gets the url of the current module, depending on where it's loaded, more about that [here](https://dmitripavlutin.com/javascript-import-meta/) and [here](https://blog.logrocket.com/alternatives-dirname-node-js-es-modules/).

Next, `new URL(input[, base])`  creates a new `URL` object by parsing the `input` relative to the `base`. If `input` is relative then `base` is required but if `input` is absolute then the `base` is ignored. Also, if `base` is passed as a string, it will be parsed equivalent to `new URL(base)` where base is if, I'm not wrong, concatenated with input. More on that [here](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL).

Finally, `fileURLToPath` decodes the file URL to a path string and makes sure that the URL control characters such as `/, %` are correctly appended/adjusted when converting the given file URL into a path. It returns a string which represents the fully-resolved platform-specific file path.

You said that you usually do this by using `__dirname` and `__filename` or something similar. Because these are not defined when using ES module, workaround with `fileURLToPath` and `import.meta.url` is used.
If I'm not wrong:
- `__filename` is equivalent to `fileURLToPath(import.meta.url)`
- `__dirname` is equivalent to `fileURLToPath(new URL('.', import.meta.url));`

I hope this at least gives you an idea about how it works. 😁

Reply:
It gives an idea 😄.. tnx for the info